staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -956,6 +960,12 @@ func (alloc *allocator) allocateDevice(r deviceIndices, device deviceWithID, mus
 		subRequestName = requestData.request.name()
 	}
 
+	// Might be tainted, in which case the taint has to be tolerated.
+	// The check is skipped if the feature is disabled.
+	if alloc.deviceTaintsEnabled && !allTaintsTolerated(device.basic, request) {
+		return false, nil, nil
```

Comment:
Not specifically about this feature though, are we not providing users with a way of knowing why each device is rejected? i.e., if a device is rejected by this feature, how could users notice that?

Reply:
This is indeed problematic and also applies to other device selection criteria and more generally to all scheduling decisions. I discussed some ideas [earlier in this PR](https://github.com/kubernetes/kubernetes/pull/130447#discussion_r2000682730).

Reply:
I think the device device allocation is tied to `ResourceClaim` resource. So, How about exposing allocation failure (I imagined "can't tolerate" case only) information on `ResourceClaim` side by events or status??

Reply:
There may be hundreds of devices that can't be used to allocate a ResourceClaim. We cannot list all of them and the reason in each case.

A single "cannot allocate" event would be possible, but also not very informative.